src/wrappers/pubsub.js

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable prefer-rest-params */
+/**
+ * @fileoverview Handlers for Pubsub instrumentation
+ */
+
+const shimmer = require('shimmer');
+const {
+    tracer,
+    tryRequire,
+} = require('epsagon');
+const { initializeEvent, finalizeEvent } = require('../runners/runner');
+const traceContext = require('../trace_context.js');
+
+const GOOGLE_CLOUD_TYPES = {
+    defaultProjectId: '{{projectId}}',
+    pubsub: {
+        name: 'pubsub',
+    },
+};
+
+const subscriber = tryRequire('@google-cloud/pubsub/build/src/subscriber');
+
+/**
+ * Handle subscriber event emitter of eventName='message'
+ * @param {Message} message received message.
+ * @param {*} originalHandler listener callback function.
+ * @param {*} requestFunctionThis request arguments.
+ */
+function pubSubSubscriberMiddleware(message, originalHandler, requestFunctionThis) {
+    try {
+        tracer.restart();
+        const functionName = originalHandler.name || 'messageHandler';
+        const { event: pubSubEvent, startTime: pubSubStartTime } = initializeEvent(
+            GOOGLE_CLOUD_TYPES.pubsub.name,
+            // eslint-disable-next-line no-underscore-dangle
+            requestFunctionThis._subscription.projectId,
+            'messagePullingListener',
+            'trigger'
+        );
+        const { event: nodeEvent, startTime: nodeStartTime } = initializeEvent(
+            'node_function', functionName, 'messageReceived', 'runner'
+        );
+        tracer.addEvent(pubSubEvent);
+        const messageData = JSON.parse(`${message.data}`);
+        const callbackResponse = { messageId: message.id, ...messageData };
```

Comment:
this is highly unstructured. put messageData under a `Message Data` key

Reply:
as we discussed yesterday, messageData is a buffer of js object.
I'm checking now with more carefully the return object from JSON.parse (if google will change this behavior)